cluster-up/cluster/kind/common.sh

```diff
@@ -141,6 +141,35 @@ function _fix_node_labels() {
     done
 }
 
+function _yq() {
+  docker run --rm -i -v ${PWD}:/workdir mikefarah/yq yq "$@"
+}
+
+function _get_cri_bridge_mtu() {
+  docker network inspect -f '{{index .Options "com.docker.network.driver.mtu"}}' bridge
+}
+
+function _patch_calico_manifest() {
+  local -r calico_manifest="$1"
+  local -r calico_diff="$2"
+  local -r destination="$3"
+
+  local -r ipip_mode=$(_yq read -d"*" "$calico_manifest" "spec.template.spec.containers.*.env.(name==CALICO_IPV4POOL_IPIP).value" | grep -Po "(Always|Never)")
+  local -r vxlan_mode=$(_yq read -d"*" "$calico_manifest" "spec.template.spec.containers.*.env.(name==CALICO_IPV4POOL_VXLAN).value" | grep -Po "(Always|Never)")
+
+  local -r cri_mtu=$(_get_cri_bridge_mtu)
+  if [ $ipip_mode == "Always" ]; then
+    overhead=$((20))
+    calico_mtu=$((cri_mtu - overhead))
+  else
+    calico_mtu=$(_yq read -d"*" "$calico_manifest" "data.veth_mtu" | grep -P "\d")
+  fi
+
+  CNI_MTU=$calico_mtu envsubst < "$calico_diff" > "/tmp/calico.diff"
```

Comment:
Congratulation, now we use all 3 options... sed, envsubst and patch.

Sed is used here: `cluster-provision/manifests/cni/calico/script.d/003-change-cidr.sh`.

Reply:
I almost sure that we can delete all files under `cluster-provision/manifests/cni/calico/script.d`, 
we used it before we started to use diff.